clients/src/main/java/org/apache/kafka/clients/producer/internals/TransactionManager.java

```diff
@@ -1072,7 +1076,11 @@ private void transitionTo(State target, RuntimeException error) {
             if (error == null)
                 throw new IllegalArgumentException("Cannot transition to " + target + " with a null exception");
             lastError = error;
+            abortableError = error;
         } else {
+            if (target != State.ABORTING_TRANSACTION) {
```

Comment:
Does this relate to the optimization https://issues.apache.org/jira/browse/KAFKA-10504?

Reply:
yes